CMakeLists.txt

```diff
@@ -67,49 +71,83 @@ if (NOT WIN32)
   set(CMAKE_CXX_STANDARD 11)
 endif()
 
-add_library(${PROJECT_NAME} STATIC
-  src/lib/DeepState.c
-  src/lib/Log.c
-  src/lib/Option.c
-  src/lib/Stream.c
-)
+if (NOT BUILD_NOSTATIC)
```

Comment:
This double negative here is a bit unusual. 

I think it's common in CMake to have `BUILD_STATIC` and `BUILD_SHARED` options, along with `BUILD_32BIT` (which might produce artifacts for both the shared and static case?). Like, maybe we want:

```cmake
option(DEEPSTATE_BUILD_SHARED "Enable build of shared libraries" ON)
option(DEEPSTATE_BUILD_STATIC "Enable build of static libraries" ON)
```

Then we could perhaps have different target names, for exmaple: https://github.com/facebook/rocksdb/blob/master/CMakeLists.txt#L724-L726 (ignoring `${ARTIFACT_SUFFIX}`) .

The options can be set at the command line invocation of cmake, with `-DDEEPSTATE_ENABLE_SHARED=ON` or `..=OFF`. Similarly, it enables another project to include our code and configure it in CMake.

@alessandrogario might have some good tips on this.

Reply:
(Agreed)